xtask/src/gendocs.rs

```diff
@@ -0,0 +1,94 @@
+//! Convert Phylum CLI to markdown.
+
+use std::path::{Path, PathBuf};
+use std::{env, fs};
+
+use anyhow::Result;
+use clap_markdown::Generator;
+use phylum_cli::app;
+
+/// Output directory.
+const OUTPUT_DIR: &str = "./docs/command_line_tool";
+
+/// Template directory.
+const TEMPLATE_DIR: &str = "./doc_templates";
+
+/// File header inserted at the top of each page.
+const HEADER: &str = "---
+title: {PH-TITLE}
+category: 6255e67693d5200013b1fa3e
+hidden: false
+---";
+
+/// Generate Phylum CLI documentation.
+pub fn gendocs() -> Result<()> {
+    // Create target directory.
+    fs::create_dir_all(Path::new(OUTPUT_DIR))?;
+
+    // Store all generated markdown pages in target directory.
+    for (path, markdown) in pages()? {
+        fs::write(path, markdown)?;
+    }
+
+    Ok(())
+}
+
+/// Get a vec with all pages and their respective paths.
+fn pages() -> Result<Vec<(PathBuf, String)>> {
+    let template_dir = Path::new(TEMPLATE_DIR);
+    let target_dir = Path::new(OUTPUT_DIR);
+
+    // Set `XDG_DATA_HOME` to a bogus directory so regardless of installed
+    // extensions, none of them are ever documented.
+    env::set_var("XDG_DATA_HOME", "/i/n/v/a/l/i/d");
```

Comment:
Now when you first look at this, you might think it's a bit of a hack. And you would be right.

But I think this is the best solution we have to just ensure no extensions are taken into account for documentation generation without actually having to modify the CLI's interface. We could add an `extensions: bool` flag to `phylum_cli::app::app`, but I don't really like the idea.

This should work just fine, even if the bogus directory exists, just as long as it's not actually where users are installing their extensions. It generates an error log message, but since we never initialize any logger nobody gets to see that anyway.

Reply:
Oh...sorry...missed this comment before I submitted my review.